### PR TITLE
Improve various documentation

### LINKED ISSUE #29: convert reST to NumPy-style docs
Need to return back and update the docs.

_Originally posted by @kne42 in https://github.com/Napari/napari-gui/pull/28/files_

### PR DESCRIPTION
**Pull request by kne42**
# Introduction
Improve various documentation across the repository.

## How Has This Been Tested?
[Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce.]
- [x] `examples/layers.ipynb`

closes #29

### CODE REVIEW

File: gui/elements/_window.py [hunk 1]
```diff
@@ -4,7 +4,12 @@
 
 
 class Window:
-    """Application window.
+    """Application window that contains the menu bar and viewers.
```
Comment (AhmetCanSolak) [inline gui/elements/_window.py:8]: menu bar, viewers, sliders and layer list maybe?

File: gui/util/range_slider.py [hunk 2]
```diff
@@ -172,30 +177,43 @@ def updateScaleValues(self):
...
 
     def getPos(self, event):
-        """
-        :param event: catches from qt context
-        :return: relative horizontal position of event
+        """Get event position.
+
+        Parameters
+        ----------
+        event : ?
```
Comment (AhmetCanSolak) [inline gui/util/range_slider.py]: qt eventloop constantly working. This methods is invoked whenever it is needed for the event of the moment. So event: Event from Qt context. That's true.

File: gui/util/range_slider.py [hunk 3]
```diff
@@ -216,39 +234,56 @@ def paintEvent(self, event):
...
 
     def rangeSliderSize(self):
-        """
-        :return: width attribute of parent class QWidget
+        """Size of the slider.
+
+        Returns
+        -------
+        size : int
+            Slider widget width.
```
Comment (AhmetCanSolak) [inline gui/util/range_slider.py]: slider could be vertical too :D saying only width could be deceptive.

**Review by AhmetCanSolak — CHANGES_REQUESTED** [review]

File: gui/util/range_slider.py (hunk 2 above)
Comment (kne42) [reply gui/util/range_slider.py]: Yeah but is there an exact `QtEvent` class or something like that?

File: gui/util/range_slider.py (hunk 3 above)
Comment (kne42) [reply gui/util/range_slider.py]: That's why the method summary is neutral in language. However, I thought that we should specify that it is the width in this case.

File: gui/util/range_slider.py (hunk 2 above)
Comment (AhmetCanSolak) [reply gui/util/range_slider.py]: There is base class `QEvent`: http://doc.qt.io/qt-5/qevent.html

File: gui/layers/__init__.py [hunk 4]
```diff
@@ -1,3 +1,10 @@
+"""Layers are the viewable objects that can be added to a viewer.
+
+Custom layers must inherit from Layer and pass along the visual node
```
Comment (jni) [inline gui/layers/__init__.py]: Can you define "visual node" here?

File: gui/elements/_window.py [hunk 5]
```diff
@@ -36,10 +41,21 @@ def add_viewer(self):
...
-    def resize(self, *args):
+    def resize(self, width, height):
+        """Resizes the window.
+
+        Parameters
+        ----------
+        width : int
+            Width in pixels.
+        height : int
+            Height in pixels.
```
Comment (jni) [inline gui/elements/_window.py]: How does Qt handle HiDPI displays?

File: gui/elements/_window.py [hunk 6]
```diff
@@ -36,10 +41,21 @@ def add_viewer(self):
...
+        """
         self._qt_window.resize(*args)
 
     def show(self):
-        self.resize(self._qt_window.layout().sizeHint())
+        """Resizes, shows, and brings forward the window.
```
Comment (jni) [inline gui/elements/_window.py]: Let's try to follow PEP257:

```
"""Resize, show, and bring forward the window."""

File: gui/layers/__init__.py (hunk 4 above)
Comment (jni) [reply gui/layers/__init__.py]: I see below that it's a vispy thing. Maybe link to the vispy docs on it?

File: gui/util/range_slider.py (hunk 3 above)
Comment (jni) [reply gui/util/range_slider.py]: Eh, I'm with @AhmetCanSolak, I think width only here could be confusing. We don't have to be so concise here, let's clarify: 

```suggestion
            Slider widget width (horizontal sliders) or height (vertical sliders).
```

**Comment by jni** [comment]
Thanks @kne42! I looooove doc updates. =) I've left some minor comments, but otherwise I'm very happy with these changes. =)

File: gui/elements/_window.py (hunk 5 above)
Comment (kne42) [reply gui/elements/_window.py]: Your guess is as good as mine, but there is some documentation available [here](http://doc.qt.io/qt-5/highdpi.html).

**Comment by kne42** [comment]
fixes #29

File: gui/elements/_window.py (hunk 5 above)
Comment (jni) [reply gui/elements/_window.py]: Ok thanks, my reading of that is, "don't worry about it, and mostly Qt will do the right thing.". =)

But, from my reading, I think this should be "height in *logical* pixels" (as opposed to *display* pixels.)

**Comment by royerloic** [comment]
Guys, what's happening here, is this resolved?

**Review by AhmetCanSolak — APPROVED** [review]

**Review by jni — APPROVED** [review]
So great now!

**Comment by jni** [comment]
So, we should either remove the Linux builds or set them to be "acceptable failures" (Travis has such a setting, not sure about Circle.), since I can't push a green button here, which is annoying.

**Comment by jni** [comment]
I take it back, it just forces you to wait for the builds. =)

**Pull request merged**